packages/supabase/src/index.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import type {
+  Adapter,
+  AdapterSession,
+  AdapterUser,
+  VerificationToken,
+} from "next-auth/adapters"
+import type { SupabaseClient } from "@supabase/supabase-js"
+import type { Account } from "next-auth"
+
+export const formatter = {
+  /** Takes a Fauna object and returns a plain old JavaScript object */
+  from<T = Record<string, unknown>>(object: Record<string, any>): T {
+    const newObject: Record<string, unknown> = {}
+    for (const key in object) {
+      const value = object[key]
+      if (key === "id" || key === "userId") {
+        newObject[key] = value.toString()
+      } else if (typeof value === "string" && Date.parse(value)) {
+        newObject[key] = new Date(value)
+      } else {
+        newObject[key] = value
+      }
+    }
+    return newObject as T
+  },
+  to(object: Record<string, any>) {
+    const newObject: Record<string, unknown> = {}
+    for (const key in object) {
+      const value = object[key]
+      if (key === "id" || key === "userId") {
+        newObject[key] = parseInt(value, 10)
+      } else {
+        newObject[key] = value
+      }
+    }
+    return newObject
+  },
+}
+
+export function SupabaseAdapter(client: SupabaseClient): Adapter {
+  const { from, to } = formatter
+  return {
+    async createUser(data) {
+      const result = await client
+        .from<AdapterUser>("users")
```

Comment:
Why do we need to create a `users` table? shouldn't we be using the `auth.users` table? ...
So in this case we can use `supabase.auth.api.createUser` to create a user in that table.

More info: https://supabase.com/docs/reference/javascript/auth-api-createuser